src/v0/destinations/redis/transform.js

```diff
@@ -35,7 +36,10 @@ const transformSubEventTypeProfiles = (message, workspaceId, destinationId) => {
   // form the hash
   const hash = `${workspaceId}:${destinationId}:${message.context.sources.profiles_entity}:${message.context.sources.profiles_id_type}:${message.userId}`;
   const key = `${message.context.sources.profiles_model}`;
-  const value = JSON.stringify(message.traits);
+  let value = JSON.stringify(message.traits);
```

Comment:
We should not JSON stringify traits by default if it is possible to not use them.

Reply:
It was being stringified before also, should I remove the stringification?

Reply:
The implementation should have looked something like this,

```
if event type is record {
  // do record specific logic
} else {
  // stringify traits
}
```

Reply:
Right now the flow is
event -
- if event is a record type event
- - handle insert/delete action related logic

- if event is having profiles' contextual data
- -  it is a profiles-identify event (backward compatible, only insertion supported)

- for none of the above
-- it is a standard redis destination's identify event

Reply:
I don't think you're getting it.

I meant that you should not stringify traits by default. Instead, do it only for non-record type of event flows.